server/src/main/java/org/elasticsearch/action/admin/indices/datastream/CreateDataStreamAction.java

```diff
@@ -165,9 +166,11 @@ static ClusterState createDataStream(ClusterState currentState, Request request)
                 throw new IllegalArgumentException("data_stream [" + request.name + "] already exists");
             }
 
+            MetaDataCreateIndexService.validateIndexOrAliasName(request.name,
+                (s1, s2) -> new IllegalArgumentException("data_stream [" + s1 + "] " + s2));
```

Comment:
The other usages of `validateIndexOrAliasName` throw a dedicated exception. I am not entirely sure if `IllegalArgumentException` results in the same RestStatus and message handling. Could we add this to the rest test suite?

Reply:
Added a REST test to enforce that the same HTTP 400 response code is returned.